dig.go

```diff
@@ -206,10 +204,33 @@ func (c *Container) isAcyclic(n node) error {
 	return detectCycles(n, c.nodes, nil)
 }
 
-// Retrieve a type from the container
-func (c *Container) get(t reflect.Type) (reflect.Value, error) {
-	if v, ok := c.cache[t]; ok {
-		return v, nil
+type keyOpt func(*key)
+
+// Specifies if absence of the object in the graph is ok on retrieval.
+func optional(opt bool) keyOpt {
```

Comment:
Optional: This pattern is probably not required here. Seems to me that `key`/`edge` expresses all the requirements for the query for a node. So you could probably just write `get(key) (reflect.Value, error)`.